pkg/ssh/ssh_keys.go

```diff
@@ -79,6 +80,29 @@ func (c *Context) GenerateSSHKey(keyName string, passphrase string) (*KeyPair, e
 	return &keyPair, nil
 }
 
+func (c *Context) GetPublicKeyFromPrivateKey(privateKeyPath string) (string, error) {
+	keygenExe, err := c.findKeygen()
+	if err != nil {
+		return "", err
+	}
+
+	keygenCmd := exec.Command(keygenExe, "-y", "-f", privateKeyPath)
```

Comment:
This commands prompts me for my private key passphrase before it prints the public key. Assuming that the public key is written to a separate `*.pub` file, couldn't we just read from the file?

Reply:
Oh I had meant to test this scenario once I got 👍 so thank you for doing the work for me 😉 

> Assuming that the public key is written to a separate *.pub file

The "assuming" part is what I was concerned about here - do you think that that is a strong enough assumption to rely on? The reason I went with `-G` in general for this spike is that I wasn't convinced we could rely on users always keeping the public key around. 

We could of course just have it as a usage rule that the user needs to have both the public and private keys around for `gh cs ssh` to work properly, but that felt overly restrictive and non-obvious from the user's perspective. I was trying to get towards "it just works" with `-G`, but this issue you've pointed out with passphrases might kill that strategy.

Reply:
>> Assuming that the public key is written to a separate *.pub file
>
> The "assuming" part is what I was concerned about here - do you think that that is a strong enough assumption to rely on? The reason I went withis approach is that I wasn't convinced we could rely on users always keeping the public key around.

`ssh-keygen` always generates a `.pub` file for every private key, so I think this is a strong-enough convention for us to rely on. Note that we don't have to necessarily come up with a fail-proof way of scanning for every possible edge case— as I understand it, this PR is about giving the user an escape hatch from the automatic `codespaces.auto` key generation, and if that escape hatch is "just have your own keypair in `~/.ssh` of which the pub key is uploaded to your GitHub account", I fully believe that's good enough!

Reply:
Fair enough 😄 I'll go with that route, it's a lot nicer to clean up the public key generation logic too!